### PR TITLE
cross compling grpc_cpp_plugin not found bug

### PR DESCRIPTION
**Pull request by seanchann**
Fixes # (issue)

## Changes

-  need use find_program on  host when CMAKE_CROSSCOMPILING enable

### CODE REVIEW

File: cmake/opentelemetry-proto.cmake [hunk 1]
```diff
@@ -84,7 +84,12 @@ foreach(IMPORT_DIR ${PROTOBUF_IMPORT_DIRS})
   list(APPEND PROTOBUF_INCLUDE_FLAGS "-I${IMPORT_DIR}")
 endforeach()
 
-set(gRPC_CPP_PLUGIN_EXECUTABLE $<TARGET_FILE:gRPC::grpc_cpp_plugin>)
+
+if(CMAKE_CROSSCOMPILING)
+    find_program(gRPC_CPP_PLUGIN_EXECUTABLE grpc_cpp_plugin)
```
Comment (ThomsonTan) [inline cmake/opentelemetry-proto.cmake:89]: Does this `find_program` only work if `grpc_cpp_plugin` is available in system search path?

Comment (seanchann) [reply cmake/opentelemetry-proto.cmake:89]: yes, but you can specify your own directory through -DCMAKE_PROGRAM_PATH=/your/path

File: cmake/opentelemetry-proto.cmake [hunk 2]
```diff
@@ -84,7 +84,12 @@ foreach(IMPORT_DIR ${PROTOBUF_IMPORT_DIRS})
...
+else()
+    set(gRPC_CPP_PLUGIN_EXECUTABLE $<TARGET_FILE:grpc_cpp_plugin>)
```
Comment (ThomsonTan) [inline cmake/opentelemetry-proto.cmake:91]: Prefix `grpc::` is missed here?

```suggestion
    set(gRPC_CPP_PLUGIN_EXECUTABLE $<TARGET_FILE:grpc::grpc_cpp_plugin>)
```

**Review by ThomsonTan — APPROVED** [review]

File: cmake/opentelemetry-proto.cmake [hunk 3]
```diff
@@ -84,7 +84,12 @@ foreach(IMPORT_DIR ${PROTOBUF_IMPORT_DIRS})
   list(APPEND PROTOBUF_INCLUDE_FLAGS "-I${IMPORT_DIR}")
 endforeach()
 
-set(gRPC_CPP_PLUGIN_EXECUTABLE $<TARGET_FILE:gRPC::grpc_cpp_plugin>)
+
+if(CMAKE_CROSSCOMPILING)
+    find_program(gRPC_CPP_PLUGIN_EXECUTABLE grpc_cpp_plugin)
+else()
+    set(gRPC_CPP_PLUGIN_EXECUTABLE $<TARGET_FILE:grpc::grpc_cpp_plugin>)
```
Comment (ThomsonTan) [inline cmake/opentelemetry-proto.cmake:91]: Need to fix casing.

```suggestion
    set(gRPC_CPP_PLUGIN_EXECUTABLE $<TARGET_FILE:gRPC::grpc_cpp_plugin>)
```

**Pull request merged**